tests/e2e/specs/api/DevfileAcceptanceTestAPI.spec.ts

```diff
@@ -3,47 +3,102 @@ import { DevWorkspaceConfigurationHelper } from '../../utils/DevWorkspaceConfigu
+                            devfilesBuildCommands.push(command);
+                        }
+                    });
+                }
+            });
 
-        kubernetesCommandLineToolsExecutor = new KubernetesCommandLineToolsExecutor(devWorkspaceName);
-        kubernetesCommandLineToolsExecutor.loginToOcp();
-    });
+            test('Create DevWorkspace', async function (): Promise<void> {
+                const devWorkspaceConfigurationYamlString: string = await devWorkspaceConfigurationHelper.getDevWorkspaceConfigurationYamlAsString(devfileContext);
+                const applyOutput: ShellString = kubernetesCommandLineToolsExecutor.applyYamlConfigurationAsStringOutput(devWorkspaceConfigurationYamlString);
 
-    test('Create DevWorkspace', async function (): Promise<void> {
-        const devWorkspaceConfigurationYamlString: string = await devWorkspaceConfigurationHelper.getDevWorkspaceConfigurationYamlAsString(devfileContext);
-        const applyOutput: ShellString = kubernetesCommandLineToolsExecutor.applyYamlConfigurationAsStringOutput(devWorkspaceConfigurationYamlString);
+                expect(applyOutput.stdout)
+                    .contains('devworkspacetemplate')
+                    .and.contains('devworkspace')
+                    .and.contains.oneOf(['created', 'configured']);
 
-        expect(applyOutput.stdout)
-            .contains('devworkspacetemplate')
-            .and.contains('devworkspace')
-            .and.contains.oneOf(['created', 'configured']);
+            });
 
-    });
+            test('Wait until DevWorkspace has status "ready"', async function (): Promise<void> {
+                expect(kubernetesCommandLineToolsExecutor.waitDevWorkspace().stdout).contains('condition met');
+            });
 
-    test('Wait until DevWorkspace has status "ready"', async function (): Promise<void> {
-        expect(kubernetesCommandLineToolsExecutor.waitDevWorkspace().stdout).contains('condition met');
-    });
+            test('Check if project was created', function (): void {
+                clonedProjectName = StringUtil.getProjectNameFromGitUrl(devfileSample.link);
+                expect(containerTerminal.ls().stdout).includes(clonedProjectName);
+            });
 
-    suiteTeardown('Delete DevWorkspace', async function (): Promise<void> {
-        kubernetesCommandLineToolsExecutor.deleteDevWorkspace();
-    });
-});
+            test('Check if files were imported', function (): void {
+                containerWorkDir = containerTerminal.pwd().stdout.replace('\n', '');
+                expect(containerTerminal.ls(`${containerWorkDir}/${clonedProjectName}`).stdout).includes(`devfile.yaml`);
+            });
 
+            test(`Check if build command returns success`, function (): void {
+                if (devfilesBuildCommands.length === 0) {
+                    Logger.info(`Devfile does not contains build commands.`);
```

Comment:
It could be mentioned earlier, in test "Collect build commands from the devfile"

Reply:
Well, script could log `Devfile does not contains build commands.` in test "Collect build commands from the devfile", and skip test`Check if build command returns success` and so make scenario one test simpler, couldn't it?

Reply:
If you are talking about `if(devfileBuildCommands === 0) this.test.skip()` it is impossible, because mocha framework initializes completely before any test code runs, but collection of commands inside another test. So we will receave skip for this test for any sample.

Reply:
I think there could be non trivial workaround, but not sure it necessary